### ISSUE TITLE
mathjax 2.5 \sqrt render problem in google chrome

### ISSUE THREAD

**Issue by hejinjun** [mathjax/MathJax#1258]
![20150906184413](https://cloud.githubusercontent.com/assets/2560843/9704017/ced4b5ca-54c8-11e5-91f6-7fd04a72c92a.png)

I found \sqrt is not render very well in google chrome and this render problem is only for the version after 2.5.

test sample:$$\sqrt[a+4b]{4a+11b}$$ and $$\sqrt[2a+6b]{a+4b+1}$$

**Comment by pkra** [comment]
I'm unable to reproduce this. Can you provide a live page that exhibits the problem? 

See also https://github.com/mathjax/MathJax/blob/master/CONTRIBUTING.md#bug-reports for more tips to write bug reports.

**Comment by dpvc** [comment]
I also can't reproduce the issue.  I suspect it is either other CSS on the page interfering with MathJax, or (more likely), the minimum font size in the browser preventing MathJax from making the fonts smaller.

In versions of MathJax prior to v2.5, the HTML-CSS output would measure every item that it had to position (like the "a+4b"), so that it was certain that the size it had matched the actual size in the browser.  That was a source of significant performance degradation, however, so in v2.5 and above MathJax no longer measures everything but uses its internal measurements for all but very specific cases.  This works well for most situations, but the case where the browser's minimum font size limits MathJax's ability to change the font size is one where the internal measurements will not match the browser's sizes properly.  So in v2.4, MathJax would have detected the changed size and compensated, but in v2.5 that is no longer the case.

You can still request the extra measurements, however, but adding

	MathJax.Hub.Config({
	  "HTML-CSS": {noReflows: false}
	});

to you configuration.  This will force the HTML-CSS render to perform the measurements as in 2.4, at the loss of some rendering speed.

Davide

**Comment by hejinjun** [comment]
live page url:http://plnkr.co/edit/H784j8Bi1Ugn5NDwaE3q?p=preview

after adding { noReflows: false } to my configuration.It render well.

ps. I also had try type my latex text to MathJax Live Demo(https://www.mathjax.org/) ,but output is also not well. Is this my chrome's problem?

![latex](https://cloud.githubusercontent.com/assets/2560843/9714973/9574434c-558e-11e5-9a49-64a1b08880ad.png)

Thx Davide & Peter !

**Comment by pkra** [comment]
> live page url:http://plnkr.co/edit/H784j8Bi1Ugn5NDwaE3q?p=preview

Thanks. I don't see any issues there either, I'm afraid (using Chrome 45/46).

> after adding { noReflows: false } to my configuration.It render well.

Glad that worked for you.

> I also had try type my latex text to MathJax Live Demo(https://www.mathjax.org/) ,but output is also not well. 

I don't see the issue there either, I'm afraid.

> Is this my chrome's problem?

Could be. It could also be an extension is interfering.

**Comment by dpvc** [comment]
It also renders properly for me.  I'm wondering if you have a minimum font size set.  In Chrome, open the settings panel (via the settings menu, or open `chrome://settings`), and click the "Show advanced settings" link at the bottom of the main part of the settings page.  That will make more settings available.  Scroll down to "Web content" and click the "Customize fonts" button.  At the bottom is a slider for the minimum font size.  Move that all the way to the left (if it isn't already) and see if that makes a difference for you.

**Comment by hejinjun** [comment]
I had upgrade my chrome to version 45 and uninstall  all  extension , also change the  "Customize fonts"  in  "Web content", but it's not really work for my windows chrome.I can easily reproduce this issues in my mate's windows OS computer.

I think this issues maybe cause by the font-family for the sup text like(a+4b).maybe the plus symbol renders too big.

I just have test this demo in MacBook chrome and it renders properly in mac chrome , when I diff the DOM elements , I found the difference between MacBook chrome and windows chrome is font-family for the latex text. 

windows chrome DOM:
![win](https://cloud.githubusercontent.com/assets/2560843/9726866/0607b0f2-562b-11e5-925a-9cf7c2090077.png)

windows chrome output:
![win1](https://cloud.githubusercontent.com/assets/2560843/9727010/9ae5d5f4-562c-11e5-84e0-506d8d6071fe.png)


mac chrome DOM:
![mac](https://cloud.githubusercontent.com/assets/2560843/9726963/02cd83c0-562c-11e5-8afe-184eab4e19a0.jpg)

mac chrome output:
![mac2](https://cloud.githubusercontent.com/assets/2560843/9727018/b1ef338a-562c-11e5-87ff-15a018e30efc.jpg)

**Comment by pkra** [comment]
> I found the difference between MacBook chrome and windows chrome is font-family for the latex text. 

Mac comes with the STIX fonts pre-installed and MathJax by default uses those, so the change you see is expected. I'm afraid that's unrelated though as we tested the example using the MathJax TeX fonts.

**Comment by dpvc** [comment]
It still looks like a minimum font size problem to me.  The font size of the a+4b is not the 50% that the style is indicating it should be.  The DOM elements are correctly set for the root, and the placement and clipping in the container elements look appropriate for a 50% size, but the image doesn't show a 50% font.  So I'd check the font settings in the "Customize fonts" panel again.  Can you send a screen shot of the slider at the bottom of that panel?

**Comment by hejinjun** [comment]
Thanks Davide , you are right , I'd check my chrome  default minimum font size. It's 12px, so 50% size also be the 12px. after adjust the default minimum font size value, It renders properly.

![well](https://cloud.githubusercontent.com/assets/2560843/9752090/f0bcd330-56de-11e5-9dee-687dcca93d81.png)

The default minimum font size is 12px in chrome(chinese version). That is why chrome does not display font which size is less than 12px in Chinese UI.  Is there a way to adjust default minimum font size by CSS or javascript? because it is unrealistic to help all clients to adjust the default minimum font size by browser setting.

**Comment by pkra** [comment]
> Is there a way to adjust default minimum font size by CSS or javascript?

No. See http://stackoverflow.com/a/12631243 for some discussion and some old bugs that could be exploited. 

You could use MathJax's SVG output which does not use fonts but SVG paths and thus is not affected by the setting. But best practice would be to revise your CSS to use larger font sizes (especially since math layout can easily have even smaller font sizes).

**Comment by hejinjun** [comment]
ok, thank you. I will try to revise my CSS.
